src/components/TextFormatter.js

```diff
@@ -0,0 +1,30 @@
+import React, { PureComponent } from 'react';
+
+export default class TextFormatter extends PureComponent {
+    state = {
+      text: '',
+      color: 'black'
+    }
```

Comment:
I imagine you wrote these two handlers to be more explicit and clear for yourself, but you can also combine them like so...
```
handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
}
```